keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,891 @@
+            use on the "top" layer. Ignored unless `include_top=True`. Set
+            `classifier_activation=None` to return the logits of the "top" layer.
+        :param include_preprocessing: Boolean, whether to include the preprocessing layer
+            (`Rescaling`) at the bottom of the network. Defaults to `True`.
+    Returns:
+        A `tf.keras.Model` instance.
+    Raises:
+        ValueError: in case of invalid argument for `weights`, or invalid input
+            shape.
+        ValueError: if `classifier_activation` is not `softmax` or `None` when
+            using a pretrained top layer.
+    """
+    # Validate parameters
+    available_weight_variants = DEPTH_TO_WEIGHT_VARIANTS[depth]
+    if weights == "imagenet":
+        max_input_shape = max(available_weight_variants)
+        """logging.warning(
+            f"Received `imagenet` argument without "
+            f"explicit weights input size. Picking weights trained with "
+            f"biggest available shape: imagenet-i{max_input_shape}"
+        )"""
+        weights = f"{weights}-i{max_input_shape}"
+
+    weights_allow_list = [f"imagenet-i{x}" for x in available_weight_variants]
+    if not (weights in {*weights_allow_list, None} or file_io.file_exists_v2(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` (random initialization), `'imagenet'` "
+            "(pre-training on ImageNet, with highest available input shape),"
+            " or the path to the weights file to be loaded. "
+            f"For ResNetRS{depth} the following weight variants are "
+            f"available {weights_allow_list} (default=highest)."
+            f" Received weights={weights}"
+        )
+
+    if weights in weights_allow_list and include_top and classes != 1000:
+        raise ValueError(
+            f"If using `weights` as `'imagenet'` or any of {weights_allow_list} with "
+            f"`include_top` as true, `classes` should be 1000. "
+            f"Received classes={classes}"
+        )
+    input_shape = imagenet_utils.obtain_input_shape(
+      input_shape,
+      default_size=224,
+      min_size=32,
+      data_format=backend.image_data_format(),
+      require_flatten=include_top,
+      weights=weights)
+    # Define input tensor
+    if input_tensor is None:
+        img_input = layers.Input(shape=input_shape)
+    else:
+        if not backend.is_keras_tensor(input_tensor):
+            img_input = layers.Input(tensor=input_tensor, shape=input_shape)
+        else:
+            img_input = input_tensor
+    bn_axis = 3 if backend.image_data_format() == "channels_last" else 1
+    x = img_input
+    if include_preprocessing:
+        num_channels = input_shape[bn_axis - 1]
+        if  num_channels == 3:
+            x = layers.Rescaling(scale=1. / 255)(x)
+            x = layers.Normalization(
+                mean=[0.485, 0.456, 0.406],
+                variance=[0.229**2, 0.224**2, 0.225**2],
```

Comment:
Why use mean-standardization here? In general `Rescaling` is enough for preprocessing. Subtracting the mean does not help the model.